File: app/src/main/java/com/ysr/express/adapter/SearchListAdapter.java
package com.ysr.express.adapter;

import android.content.Context;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.google.gson.Gson;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import com.google.gson.reflect.TypeToken;
import com.ysr.express.R;
import com.ysr.express.bean.ComDataBean;
import com.ysr.express.bean.RequestShipperName;
import com.ysr.library.utils.ComUtils;
import com.ysr.library.utils.LogUtils;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by ysr on 2017/7/13 12:34.
 * 邮箱 ysr200808@163.com
 */

public class SearchListAdapter extends RecyclerView.Adapter {
    private Context context;
    private List<RequestShipperName.ShippersBean> mData;
    private onItemClickListener itemClickListener;
    private ComDataBean userBean;
    private ArrayList<ComDataBean> beanList;

    public void setData(List<RequestShipperName.ShippersBean> data) {
        mData = data;
    }

    public void update(List<RequestShipperName.ShippersBean> list) {
        this.mData = list;
        this.notifyDataSetChanged();
    }

    public void setItemClickListener(onItemClickListener itemClickListener) {
        this.itemClickListener = itemClickListener;
    }

    public SearchListAdapter(Context context, List<RequestShipperName.ShippersBean> mData) {
        this.context = context;
        this.mData = mData;
        getImageUrl();
    }

    //解析快递公司图标Json
    private void getImageUrl() {
        String result = ComUtils.getJson(context, "CompanyData.json");
        //先转JsonObject
        JsonObject jsonObject = new JsonParser().parse(result).getAsJsonObject();
        //再转JsonArray 加上数据头
        JsonArray jsonArray = jsonObject.getAsJsonArray("company_data");
        Gson gson = new Gson();

        beanList = new ArrayList<>();
        //循环遍历
        for (JsonElement user : jsonArray) {
            //通过反射 ComDataBean.class
            userBean = gson.fromJson(user, new TypeToken<ComDataBean>() {
            }.getType());
            beanList.add(userBean);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_search_info, null));
    }

    @Override
    public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int position) {
        final int adjPosition = position;
        ViewHolder viewHolder = (ViewHolder) holder;
        viewHolder.tv_Courier_info.setText(mData.get(position).getShipperName());

        for (int i = 0; i < beanList.size(); i++) {
            for (int j = 0; j < beanList.get(i).getCompany().size(); j++) {
                if (mData.get(position).getShipperCode().equals(beanList.get(i).getCompany().get(j).getCode())) {
                    LogUtils.loge(beanList.get(i).getCompany().get(j).getLogo());
                    Glide.with(context)
                            .load("https://www.kuaidi100.com/images/all/" + beanList.get(i).getCompany().get(j).getLogo())
                            .into(viewHolder.image_express);
                }
            }
        }

        if (null != itemClickListener) {
            viewHolder.cv_click.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Object tag = v.getTag();
                    if (tag instanceof RequestShipperName.ShippersBean) {
                        itemClickListener.onItemTextClick(holder.itemView, adjPosition, (RequestShipperName.ShippersBean) tag);
                    }

                }
            });
            RequestShipperName.ShippersBean one = mData.get(adjPosition);
            if (null != one) {
                ((ViewHolder) holder).cv_click.setTag(one);
            }
        }


    }

    @Override
    public int getItemCount() {
        return mData == null ? 0 : mData.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public CardView cv_click;
        public ImageView image_express;
        public TextView tv_Courier_info;

        public ViewHolder(View itemView) {
            super(itemView);
            cv_click = itemView.findViewById(R.id.cv_click);
            tv_Courier_info = itemView.findViewById(R.id.tv_Courier_info);
            image_express = itemView.findViewById(R.id.image_express);
        }
    }

    /**
     * 点击事件选择回调
     */
    public interface onItemClickListener {
        void onItemTextClick(View view, int position, RequestShipperName.ShippersBean tag);
    }

}

File: app/src/main/java/com/ysr/express/adapter/SearchListAdapter.kt
package com.ysr.express.adapter

import android.content.Context
import android.support.v7.widget.CardView
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.bumptech.glide.load.engine.DiskCacheStrategy
import com.bumptech.glide.request.RequestOptions
import com.google.gson.Gson
import com.google.gson.JsonParser
import com.google.gson.reflect.TypeToken
import com.ysr.express.R
import com.ysr.express.bean.ComDataBean
import com.ysr.express.bean.RequestShipperName
import com.ysr.express.retrofit.API
import com.ysr.library.utils.ComUtils
import java.util.*

/**查询结果Adapter
 * Created by ysr on 2017/7/13 12:34.
 * 邮箱 ysr200808@163.com
 */

class SearchListAdapter(val context: Context, var mData: List<RequestShipperName.ShippersBean>?) : RecyclerView.Adapter<SearchListAdapter.ViewHolder>() {
    private var itemClickListener: onItemClickListener? = null
    private var userBean: ComDataBean? = null
    private var beanList: ArrayList<ComDataBean>? = null
    val options = RequestOptions()
            .centerCrop()
            .placeholder(R.mipmap.app_res_images_ic_no_exist)
//                        .skipMemoryCache(true)
            .error(R.mipmap.app_res_images_ic_no_exist)
            .diskCacheStrategy(DiskCacheStrategy.NONE)!!

    init {
        getImageUrl()
    }

    fun setData(list: List<RequestShipperName.ShippersBean>) {
        mData = list
    }

    fun update(list: List<RequestShipperName.ShippersBean>) {
        this.mData = list
        this.notifyDataSetChanged()
    }

    fun setItemClickListener(itemClickListener: onItemClickListener) {
        this.itemClickListener = itemClickListener
    }


    //解析快递公司图标Json
    fun getImageUrl() {
        val result = ComUtils.getJson(context, "CompanyData.json")
        //先转JsonObject
        val jsonObject = JsonParser().parse(result).asJsonObject
        //再转JsonArray 加上数据头
        val jsonArray = jsonObject.getAsJsonArray("company_data")
        beanList = ArrayList<ComDataBean>()
        //循环遍历
        for (user in jsonArray) {
            //通过反射 ComDataBean.class
            userBean = Gson().fromJson<ComDataBean>(user, object : TypeToken<ComDataBean>() {

            }.type)
            beanList!!.add(userBean!!)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchListAdapter.ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_search_info, null))
    }

    override fun onBindViewHolder(holder: SearchListAdapter.ViewHolder, position: Int) {
        val adjPosition = position
        val viewHolder = holder
        viewHolder.tv_Courier_info.text = mData!![position].ShipperName

        loop@ for (i in beanList!!.indices) {

            for (j in 0..beanList!![i].company!!.size){
                if (mData!![position].ShipperCode == beanList!![i].company!![j].code) {
                    Glide.with(context)
                            .load(API.LogoBaseUrl + beanList!![i].company!![j].logo!!)
                            .apply(options)
                            .into(viewHolder.image_express)
                    //终止此次循环
                    break@loop
                } else {
                    Glide.with(context)
                            .load(R.mipmap.app_res_images_ic_no_exist)
                            .into(viewHolder.image_express)
                }
            }
        }
        if (null != itemClickListener) {
            viewHolder.cv_click.setOnClickListener { v ->
                val tag = v.tag
                if (tag is RequestShipperName.ShippersBean) {
                    itemClickListener!!.onItemTextClick(holder.itemView, adjPosition, tag)
                }
            }
            val one = mData!![adjPosition]
            if (null != one) {
                holder.cv_click.tag = one
            }
        }


    }

    override fun getItemCount(): Int {
        return if (mData == null) 0 else mData!!.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var cv_click: CardView = itemView.findViewById<CardView>(R.id.cv_click)
        var image_express: ImageView = itemView.findViewById<ImageView>(R.id.image_express)
        var tv_Courier_info: TextView = itemView.findViewById<TextView>(R.id.tv_Courier_info)

    }

    /**
     * 点击事件选择回调
     */
    interface onItemClickListener {
        fun onItemTextClick(view: View, position: Int, tag: RequestShipperName.ShippersBean)
    }

}
